openjdk/mmtkVMCompanionThread.hpp

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 1998, 2016, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#ifndef MMTK_OPENJDK_MMTK_VM_COMPANION_THREAD_HPP
+#define MMTK_OPENJDK_MMTK_VM_COMPANION_THREAD_HPP
+
+#include "mmtkVMOperation.hpp"
+#include "runtime/mutex.hpp"
+#include "runtime/perfData.hpp"
+#include "runtime/thread.hpp"
+#include "runtime/vmOperations.hpp"
+
+class MMTkVMCompanionThread: public NamedThread {
+public:
+  enum stw_state {
+    _threads_suspended,
+    _threads_resumed,
+  };
+private:
+  Monitor* _lock;
+  stw_state _desired_state;
+  stw_state _reached_state;
+  size_t _resumption_count;
```

Comment:
This count is maintained but not used.

Reply:
I once considered moving the `start_the_world_count` into the companion thread as `_resumption_count`, but it didn't work as expected.  I will remove `_resumption_count`.